### PR TITLE
bring removeduplicates to plugin

### PR DESCRIPTION
**Pull request by JaneLiuL**

### CODE REVIEW

**Comment by knelasevero** [comment]
I think the deletiion of [duplicates.go](https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/duplicates.go) and [duplicates_test.go](https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/duplicates_test.go)  are missing. With them moved to the framework better to delete from `pkg/descheduler/strategies/`

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates.go:42]: s/strategy/plugin

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates.go:45]: s/strategy/plugin

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates.go:54]: RemoveDuplicates is one of the plugins that are aiming for balancing pods among nodes. Making [BalancePlugin](https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/framework/types.go#L69) interface more suitable.

s/DeschedulePlugin/BalancePlugin.

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates.go:127]: Better to check for `len(ownerRefList) == 0` first since `hasExcludedOwnerRefKind` is computationally more expensive.

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Thank You @JaneLiuL. I have found only few nits. Also as mentioned by @knelasevero it would be great to remove both
https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/duplicates.go and https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/duplicates_test.go in the same commit. So we can see the difference of both files and compare it with the new plugin files.

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (JaneLiuL) [reply pkg/framework/plugins/removeduplicates/removeduplicates.go]: done

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (JaneLiuL) [reply pkg/framework/plugins/removeduplicates/removeduplicates.go]: done~

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (JaneLiuL) [reply pkg/framework/plugins/removeduplicates/removeduplicates.go]: done

File: pkg/framework/plugins/removeduplicates/removeduplicates.go
Comment (JaneLiuL) [reply pkg/framework/plugins/removeduplicates/removeduplicates.go]: i move len before hasExcludedOwnerRefKind, would you have a look?

File: pkg/apis/componentconfig/validation/validation_pluginargs.go
Comment (ingvagabund) [inline pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: @a7i would you be fine keeping the validation as it is and refactoring the code after `ValidateNamespacesArgs` is introduced?

File: pkg/framework/plugins/removeduplicates/removeduplicates_test.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates_test.go:790]: Nit: `evictions.WithNodeFit(false)` can be dropped as the default value is false

File: pkg/framework/plugins/removeduplicates/removeduplicates_test.go
Comment (ingvagabund) [inline pkg/framework/plugins/removeduplicates/removeduplicates_test.go:200]: All `excludeOwnerKinds:       []string{},` lines can be dropped as `[]string{}` is the default value.

File: test/e2e/e2e_duplicatepods_test.go
Comment (ingvagabund) [inline test/e2e/e2e_duplicatepods_test.go:168]: `evictions.WithNodeFit(false)` can be dropped as well

**Comment by ingvagabund** [comment]
@JaneLiuL few more nits which are more of a cosmetic nature. I will leave the lgtm part for @a7i wrt. https://github.com/kubernetes-sigs/descheduler/pull/892#discussion_r938631980 comment.

/approve

**Comment by JaneLiuL** [comment]
thanks @ingvagabund  may i update at next pr? i will bring the toomayrestarts plugin tonight after this pr merge.

File: pkg/apis/componentconfig/validation/validation_pluginargs.go
Comment (JaneLiuL) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: i can bring this at toomanyrestart plugins pr at tonight. :)

Comment (a7i) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: Yes, happy to.

Comment (a7i) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: https://github.com/kubernetes-sigs/descheduler/pull/899

Comment (JaneLiuL) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: so i need to wait #899 merge and rebase?

Comment (a7i) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: Up to you.  I can rebase my PR if you end up merging first

Comment (JaneLiuL) [reply pkg/apis/componentconfig/validation/validation_pluginargs.go:54]: Ok,this pr still lack of lgtm label, so if your pr merge, i will rebase

**Comment by JaneLiuL** [comment]
@ingvagabund would you help to have a look for this pr? :)

**Comment by ingvagabund** [comment]
`ValidateRemoveDuplicatesArgs` built on top of https://github.com/kubernetes-sigs/descheduler/pull/899

/lgtm

**Pull request merged**
